src/Xamarin.Android.Build.Tasks/Tasks/StripNativeLibraries.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Diagnostics;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Android.Build.Tasks;
+using Xamarin.Android.Tools;
+
+namespace Xamarin.Android.Tasks
+{
+	/// <summary>
+	/// Strips debug information from the native libraries
+	/// </summary>
+	public class StripNativeLibraries : AndroidTask
+	{
+		public override string TaskPrefix => "SNL";
+
+		[Required]
+		public ITaskItem [] Libraries { get; set; }
+
+		[Required]
+		public string LocalPath { get; set; }
+
+		[Required]
+		public string AndroidBinUtilsDirectory { get; set; }
+
+		[Output]
+		public ITaskItem [] OutputLibraries { get; set; }
+
+		public override bool RunTask ()
+		{
+			if (Libraries == null || Libraries.Length == 0)
+				return true;
+
+			var output = new List<ITaskItem> (Libraries.Length);
+			var ext = OS.IsWindows ? ".exe" : "";
+			foreach (var library in Libraries) {
+				var abi = AndroidRidAbiHelper.GetNativeLibraryAbi (library);
+				if (string.IsNullOrEmpty (abi)) {
+					var packageId = library.GetMetadata ("NuGetPackageId");
+					if (!string.IsNullOrEmpty (packageId)) {
+						Log.LogCodedWarning ("XA4301", library.ItemSpec, 0, Properties.Resources.XA4301_ABI_NuGet, library.ItemSpec, packageId);
+					} else {
+						Log.LogCodedWarning ("XA4301", library.ItemSpec, 0, Properties.Resources.XA4301_ABI, library.ItemSpec);
+					}
+					continue;
+				}
+
+				var tripple = GetNdkTrippleFromAbi (abi);
+				var exe = Path.Combine (AndroidBinUtilsDirectory, $"{tripple}-strip{ext}");
+				var localDir = Path.Combine (LocalPath, library.GetMetadata ("RuntimeIdentifier"));
+				Directory.CreateDirectory (localDir);
+
+				var filename = library.GetMetadata ("Filename");
+				var extension = library.GetMetadata ("Extension");
+				var localLib = Path.Combine (localDir, $"{filename}{extension}");
+				using var proc = Process.Start (Path.Combine (AndroidBinUtilsDirectory, $"{tripple}-strip{ext}"), $"--strip-debug \"{library.ItemSpec}\" -o \"{localLib}\"");
```

Comment:
no brackets? Is that a thing now 🤔

Reply:
Do we ever plan to run these in parallel?

If it's a single `Process.Start()`, can this be a [ToolTask](https://docs.microsoft.com/en-us/dotnet/api/microsoft.build.utilities.tooltask?view=msbuild-16-netcore) instead? (We have `AndroidToolTask` to inherit from).

Reply:
yup, https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/using#using-declaration

Reply:
I guess we can run them in parallel if we care enough at some point.

Reply:
We should probably use `ToolTask` then, as it will handle errors for us.

If it prints something like `error: foo`  or `warning: foo`, it will emit the right MSBuild error/warning.